core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3537,110 +3537,90 @@ class ViewMapping<
 namespace Kokkos {
 namespace Impl {
 
-template <unsigned, class MapType>
-KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType&) {
-  return true;
+template <class Map, class... Indices, std::size_t... Enumerate>
+KOKKOS_FUNCTION bool check_bounds(Map const& map,
+                                  std::index_sequence<Enumerate...>,
+                                  Indices... indices) {
+  return (... && ((std::size_t)indices < map.extent(Enumerate)));
 }
 
-template <unsigned R, class MapType, class iType, class... Args>
-KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType& map,
-                                                        const iType& i,
-                                                        Args... args) {
-  return (size_t(i) < map.extent(R)) &&
-         view_verify_operator_bounds<R + 1>(map, args...);
+template <class... Indices>
+KOKKOS_FUNCTION constexpr char* append_formated_multidimensional_index(
+    char* dest, Indices... indices) {
+  char* d = dest;
+  Kokkos::Impl::strcat(d, "[");
+  (
+      [&] {
+        d += Kokkos::Impl::strlen(d);
+        (void)Kokkos::Impl::to_chars_i(d,
+                                       d + 20,  // 20 digits ought to be enough
+                                       indices);
+        Kokkos::Impl::strcat(d, ",");
+      }(),
+      ...);
```

Comment:
Why not just
```suggestion
      for(index: {indices...}) {
        d += Kokkos::Impl::strlen(d);
        Kokkos::Impl::to_chars_i(d,
                                       d + 20,  // 20 digits ought to be enough
                                       index);
        Kokkos::Impl::strcat(d, ",");
      }
```
for a little more "standard" syntax. If we really want to allow different types we can cast `indices` in the initializer list to `std::size_t` or so.

Reply:
Also the current implementation would yield
```C++
  Kokkos::View<int**> v("v",2,2);
  std::cout << v(1u, -1);  // Kokkos::View ERROR: out of bounds access label=("v") with indices [1,-1] but extents [2,2]
```
which is better than `Kokkos::View ERROR: out of bounds access label=("v") with indices [1,18446744073709551615] but extents [2,2]`

Reply:
agreed